invenio_accounts/api.py

```diff
@@ -8,25 +8,60 @@
 
 """API objects for Invenio Accounts."""
 
-from collections import defaultdict
+
+class Session:
```

Comment:
From what @zzacharo mentioned in the call:

I've checked uwsgi docs, shared mem is not thread safe (sometimes they do magic with the GIL). But in this case is not, so we might need to make the whole session object thread safe (e.g. locks)?

Reply:
I see ... This is meant to be used by signal custom handlers and they look like they are thread safe (see [here docs](https://docs.python.org/3.4/library/signal.html#execution-of-python-signal-handlers)).

Reply:
> I see ... This is meant to be used by signal custom handlers and they look like they are thread safe (see [here docs](https://docs.python.org/3.4/library/signal.html#execution-of-python-signal-handlers)).

I am not sure that I understand this very well. There is nothing preventing the usage of this `Session` in other ways.
Why do we need to store different `session_id`? Let's chat IRL.

Reply:
@ppanero we double checked, [blinker](https://blinker.readthedocs.io/en/stable/#blinker-documentation) is thread-safe.